pkg/cmd/roachtest/tests/restore.go

```diff
@@ -680,6 +700,34 @@ func (sp *restoreSpecs) initRestorePerfMetrics(
 	}
 }
 
+// checkFingerprint runs a stripped fingerprint on all user tables in the cluster if the restore
+// spec has a nonzero fingerprint.
+func (sp *restoreSpecs) checkFingerprint(ctx context.Context) {
+	if sp.fingerprint == 0 {
+		sp.t.L().Printf("Fingerprint not found in specs. Skipping fingerprint check.")
+		return
+	}
+
+	conn, err := sp.c.ConnE(ctx, sp.t.L(), sp.c.Node(1)[0])
+	require.NoError(sp.t, err)
+	sql := sqlutils.MakeSQLRunner(conn)
+
+	var minUserTableID, maxUserTableID uint32
+	sql.QueryRow(sp.t, `SELECT min(id) FROM system.namespace WHERE "parentID" >1`).Scan(&minUserTableID)
+	sql.QueryRow(sp.t, `SELECT max(id) FROM system.namespace WHERE "parentID" >1`).Scan(&maxUserTableID)
```

Comment:
nit: can scan into both min and max ints here at the same time with `SELECT min(id), max(id) FROM ... .Scan(&minUserTableID, &maxUserTableID)`

Reply:
Done.